File: algorithms/0049.Group Anagrams/solution.go
package main

func groupAnagrams(strs []string) [][]string {
	groupMap := make(map[[26]int][]string)
	for _, str := range strs {
		var counts [26]int
		for _, char := range str {
			counts[char-'a']++
		}
		groupMap[counts] = append(groupMap[counts], str)
	}
	var result [][]string
	for _, group := range groupMap {
		result = append(result, group)
	}
	return result
}

File: algorithms/0049.Group Anagrams/solution.py
from typing import List, Dict

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groupMap: Dict[str, List[str]] = {}
        for s in strs:
            counts = [0] * 26
            for c in s:
                counts[ord(c) - 97] += 1
            key = ','.join(map(str, counts))
            groupMap[key] = groupMap.get(key, []) + [s]
        return list(groupMap.values())
